src/vfb_connect/cross_server_tools.py

```diff
@@ -43,26 +71,137 @@ def get_terms_by_region(self, region, cells_only=False, verbose=False, query_by_
+        terms = self.oc.get_superclasses("%s" % class_expression, query_by_label=query_by_label)
         return self.neo_query_wrapper.get_type_TermInfo(list(map(gen_short_form, terms)))
 
-    def get_images(self, class_expression, query_by_label = True, direct = False):
-        """Generate JSON report of all images of the submitted type."""
+    def get_instances(self, class_expression, query_by_label=True, direct=False):
+        """Generate JSON report of all instances of class_expression. Instances are specific examples
+        of a type/class of structure, e.g. a specific instance of the neuron DA1 adPN from the FAFB_catmaid
+         database.  Instances are typically associated with registered 3D image data and may include
+         connectomics data."""
         if not re.search("'", class_expression):
             class_expression = "'" + class_expression + "'"
         terms = self.oc.get_instances("%s" % class_expression, query_by_label=query_by_label)
         return self.neo_query_wrapper.get_anatomical_individual_TermInfo(list(map(gen_short_form, terms)))
 
+    def _get_neurons_connected_to(self, neuron, threshold, direction, classification=None, query_by_label=True):
+        instances = []
+        directions = ['upstream', 'downstream']
+        if not (direction in directions):
+            raise Exception(ValueError)  # Needs improving
+        if classification:
+            instances = self.oc.get_instances(classification, query_by_label=query_by_label)
+        cypher_query = 'MATCH (upstream:Neuron)-[r:synapsed_to]->(downstream:Neuron) ' \
+                       'WHERE r.weight[0] > %d ' % threshold
+        if query_by_label:
+            cypher_query += 'AND %s.label = "%s" ' % (direction, neuron)
+        else:
+            cypher_query += 'AND %s.short_form = "%s" ' % (direction, neuron)
+        if classification and instances:
+            directions.remove(direction)
+            cypher_query += "AND %s.iri IN %s " % (directions[0], str(instances))
+        cypher_query += "RETURN upstream.short_form as query_neuron_id, upstream.label as query_neuron_name, r.weight as weight, " \
+                        "downstream.short_form as target_neuron_id, downstream.label as target_neuron_name"
+        r = self.nc.commit_list([cypher_query])
+        dc = dict_cursor(r)
+        return pd.DataFrame.from_records(dc)
+
+    def get_neurons_downstream_of(self, neuron, threshold, classification=None, query_by_label=True):
+        """Get all neurons downstream of individual `neuron` (short_form if query_by_label=False, otherwise label)
+        with connection strength > threshold.  Optionally restrict target neurons to those specified by
+        `classification = 'class expression' e.g. "'Kenyon cell'" or "'neuron' that overlaps 'lateral horn'"."""
+        return self._get_neurons_connected_to(neuron=neuron, threshold=threshold, direction='upstream',
```

Comment:
Not sure on how direction works in_get_neurons_connected_to but is upstream right for downstream?